Common: 유효성 응답 모델에 "number:~" 타입 프로퍼티 모델 추가

merge-simpson opened the pull request:
<!--
<br /><br /><br /><br /><br />

<table align="center" border="30">
<tr>
  <td>🏗️ 선행 작업 병합 후, 리베이스할 예정입니다.</td>
</tr>
</table>

<br /><br /><br /><br /><br />
그니까 아직 리뷰하지 말라니깐용.
<br /><br /><br /><br /><br />
<br /><br /><br /><br /><br />
-->

> 이 PR을 리뷰할 경우 #326 PR을 리뷰하지 않아도 됩니다.

## Issues

- Resolves #328 
- Resolves #329 
- Resolves #330 

<br />

## Description

- 숫자 데이터의 유효성 모델을 추가합니다.
- Rel: #323 

<br />

## Review Points

- 노션 스펙을 참고하실 수 있습니다. (로그인 필요)  
  https://www.notion.so/nettee/Auth-User-API-25001973b23a800c9e9fe2392aa7fa97#25001973b23a802aad45ea139d9f1743
- 관련 파일을 참고하실 수 있습니다.  
  - common/src/main/java/nettee/common/validation/model/interfaces/**BaseValidationProperty**.java  
    ```java
    public interface BaseValidationProperty {
        String type();
        Boolean required();
        Map<String, String> messages();
        
        // ... (중략)
    }
    ```  
    ```java
    public interface ValueBoundedValidationProperty extends BaseValidationProperty {
        Number min(); // accepting Integer, Double, BigDecimal (우리가 다루는 세 타입은 포함됨.)
        Number max(); // accepting Integer, Double, BigDecimal (우리가 다루는 세 타입은 포함됨.)
    }
    ```  
    ```java
    public interface InclusiveValueBoundedValidationProperty extends ValueBoundedValidationProperty {
        List<Boolean> inclusive();
    }
    ```  
  - common/src/main/java/nettee/common/validation/model/**ValidationResponseModel**.java

<br />

## How Has This Been Tested?

- 테스트 생략

<br />

## Additional Notes

### 다이어그램

```mermaid
classDiagram
    class BaseValidationProperty {
        &lt;&lt;interface&gt;&gt;
        +type() String
        +required() Boolean
        +messages() Map~String,String~
    }

    class ValueBoundedValidationProperty {
        &lt;&lt;interface&gt;&gt;
        +min() Long
        +max() Long
    }

    class InclusiveValueBoundedValidationProperty {
        &lt;&lt;interface&gt;&gt;
        +inclusive() List~Boolean~
    }
    
    class NumberIntValidationProperty {
        &lt;&lt;record&gt;&gt;
        +type() String
        +required() Boolean
        +messages() Map~String,String~
        +min() Long
        +max() Long
    }
    
    class NumberDecimalValidationProperty {
        &lt;&lt;record&gt;&gt;
        +type() String
        +required() Boolean
        +messages() Map~String,String~
        +min() Long
        +max() Long
    }
    
    class NumberBigDecimalValidationProperty {
        &lt;&lt;record&gt;&gt;
        +type() String
        +required() Boolean
        +messages() Map~String,String~
        +min() Long
        +max() Long
    }

    BaseValidationProperty <|-- ValueBoundedValidationProperty
    ValueBoundedValidationProperty <|-- InclusiveValueBoundedValidationProperty
    ValueBoundedValidationProperty <|.. NumberIntValidationProperty
    InclusiveValueBoundedValidationProperty <|.. NumberDecimalValidationProperty
    InclusiveValueBoundedValidationProperty <|.. NumberBigDecimalValidationProperty
```

merge-simpson committed:
feat(common): add `ValueBoundedValidationProperty` interface for numeric types validation model

Introduce `ValueBoundedValidationProperty` extending `BaseValidationProperty` to support `min` and `max` attributes for numeric types validation.

merge-simpson committed:
feat(common): add "number:int" validation property model

introduce `NumberIntValidationProperty` implementing `ValueBoundedValidationProperty`. It asserts/normalizes `type` to `"number:int"`, validates non-negative bounds with `min <= max`, initializes the messages map when needed, and fills default messages for `required`, `min`, and `max` using reserved field names. The messages map is finalized as an unmodifiable copy.

merge-simpson committed:
feat(common): add inclusive value-bounded validation property interface for numeric validation model

Introduce `InclusiveValueBoundedValidationProperty` extending `ValueBoundedValidationProperty`.
Adds `List<Boolean> inclusive()` to indicate lower/upper bound inclusivity in order `[min, max]`, enabling precise range semantics for number-based validations.

merge-simpson committed:
feat(common): add "number:decimal" validation property model

implement `NumberDecimalValidationProperty` for validation response model; implement `InclusiveValueBoundedValidationProperty`; validate non-negative bounds with `min <= max`; support `[min, max]` inclusivity; default `required` to false and `inclusive` to `[true, true]`; initialize and populate messages for `required`, `min`, `max`; finalize messages as unmodifiable map.

> Note: The current code asserts/normalizes `type` to `"number:int"`. If this model targets decimals, update the assertion and default to `"number:decimal"`.

merge-simpson committed:
feat(common): add "number:bigdecimal" validation property model

implement `NumberBigDecimalValidationProperty` for validation response model; implement `InclusiveValueBoundedValidationProperty`; validate non-negative bounds with `min <= max`; support `[min, max]` inclusivity; default `required` to false and `inclusive` to `[true, true]`; initialize and populate messages for `required`, `min`, `max`; finalize messages as unmodifiable map.

merge-simpson committed:
fix(common): correct type to "number:decimal" or "number:bigdecimal" in validation property models

merge-simpson committed:
feat(common): add jackson config for `BigDecimal` string serialization

Introduce `CommonJacksonConfig` to serialize `BigDecimal` as string in JSON response.
Deserialization still accepts both numeric and string inputs via Jackson default behavior.
TODO: Discuss whether to keep this in common module or relocate later.

wowddok99 reviewed (approved):
올려주신 코드 잘 봤습니다! 
문제있는 부분은 없는듯 하고 바로 병합하셔도 될 것 같습니다.
작업하시느라 고생하셨어요😄

merge-simpson committed:
fix(common): correct message initializing condition in numeric validation property models

```diff
-if (~ && messages.containsKey(key)) {
+if (~ && !messages.containsKey(key)) {
    message.put(key, "default message");
}
```
